### PR TITLE
LogWrapper interface created - an interface that wraps log calls to abstract the use of log calls throughtout the application. LogWrapperFactory - Creates a LogWrapper instance. Used to control the LogWrapper-s creation in a single place with a unified API (create(string tag))  NativeLogWrapper - An implementation of the LogWrapper interface using Android's native logger.

### PR DESCRIPTION
**Pull request by BarAgam**
LogWrapper interface created - an interface that wraps log calls to abstract the use of log calls throughtout the application.
LogWrapperFactory - Creates a LogWrapper instance. Used to control the LogWrapper-s creation in a single place with a unified API (create(string tag))
 NativeLogWrapper - An implementation of the LogWrapper interface using Android's native logger.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/36" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapper.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    void v(String message);
+    void v(String message, Throwable tr);
+
+    void w(String message);
+    void w(String message, Throwable tr);
+
+    void userInteraction(String message);
+
+    void onCreate();
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapper.java:26]: make sure we didnt miss anthing. go over the life cycle of activities and fragments (onAttach)

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapper.java:26]: !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/common/logging/NativeLogWrapper.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+import android.util.Log;
+
+/**
+ * Wraps the native android logger and delegates logging calls to it.
+ * Special logging methods are logged with DEBUG verbosity with a suitable prefix
+ */
+public class NativeLogWrapper implements LogWrapper {
+
+    private static final String USER_INTERACTION_PREFIX = "USER INTERACTION";
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/NativeLogWrapper.java]: why uppercase? add also ": "

File: app/src/main/java/com/teamagam/gimelgimel/app/common/logging/NativeLogWrapper.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    private static final String LIFE_CYCLE_PREFIX = "Life Cycle";
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/NativeLogWrapper.java:12]: same + ": "

File: app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapperFactory.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.teamagam.gimelgimel.app.common.logging;
+
+/**
+ * A simple factory class to enable future different concrete implementations
+ * to different tags and an easy way to change the LogWrapper implementation throughout the app.
+ */
+public class LogWrapperFactory {
+    public static LogWrapper create(String tag) {
+        return new NativeLogWrapper(tag);
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapperFactory.java:9]: dont you want to use something like that: 
private Logger(Class clazz, boolean useSimpleName) {
        this.tag = getTag(clazz);
        if (useSimpleName) {
            this.prefixMsg = getPrefixMessage(clazz.getSimpleName());
        } else {
            this.prefixMsg = getPrefixMessage(clazz.getName());
        }
    }

(from: https://github.com/oronno/log4android/blob/master/src/main/java/com/oneous/log4android/Logger.java)

Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapperFactory.java:9]: Ehum... nope. I dont see how this functionality helps

File: app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapper.java (hunk 1 above)
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/LogWrapper.java:26]: !resolved

**Pull request merged**